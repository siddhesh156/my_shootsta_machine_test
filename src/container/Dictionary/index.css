.main-container{
    margin: 5% 20%;
    display: flex;
    flex-direction: column;
  }

  h1{ 
    color: #7c795d; 
    font-family: 'Trocchi', serif; font-size: 45px; font-weight: normal; line-height: 48px; margin-bottom: 5%; 
  }
  
  @media screen and (max-width: 750px) {
    .main-container {
        margin: 5% 5%;
    }
  }