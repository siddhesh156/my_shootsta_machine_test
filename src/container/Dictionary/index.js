import React, { useState } from "react";
import { Button, InputGroup, FormControl, Card } from "react-bootstrap";

const Dictionary = () => {
  const [word, setWord] = useState("");
  const [searchData, setSearchData] = useState([]);

  const onHandleSearch = () => {
    //Get data from API
    fetch(`https://api.dictionaryapi.dev/api/v2/entries/en/${word}`)
      .then((res) => res.json())
      .then(
        (result) => {
          //Setting data from API to the state
          setSearchData(result[0].meanings[0].definitions);
        },
        (error) => {
          //Error from API
          console.log("Error from API ", error);
        }
      );
  };

  try {
    return (
      <React.Fragment>
        <div className="main-container">
          <h1>Dictionary</h1>
          <InputGroup className="mb-3">
            <FormControl
              placeholder="Search"
              aria-label="Search"
              aria-describedby="basic-addon1"
              value={word}
              onChange={(event) => setWord(event.target.value)}
            />
          </InputGroup>

          <Button
            variant="dark"
            style={{ alignSelf: "center", width: "70%" }}
            onClick={() => onHandleSearch()}
          >
            Submit
          </Button>

          {searchData.length > 0 && (
            <div className="mt-5">
              <Card style={{ background: "#7c795d" }}>
                <Card.Body
                  style={{
                    color: "rgb(247, 154, 12)",
                    fontFamily: "Trocchi, serif",
                  }}
                >
                  <Card.Title style={{ textAlign: "center" }}>
                    Definitions
                  </Card.Title>
                  <ul>
                    {searchData.map((data, index) => {
                      let checkExample = data?.example !== undefined;
                      return (
                        <li key={index}>
                          {data.definition}
                          {checkExample
                            ? " For example " + data?.example
                            : null}
                        </li>
                      );
                    })}
                  </ul>
                </Card.Body>
              </Card>
            </div>
          )}
        </div>
        <style>{`
      .main-container{
        margin: 5% 20%;
        display: flex;
        flex-direction: column;
      }

      h1{ 
        color: #7c795d; 
        font-family: 'Trocchi', serif; font-size: 45px; font-weight: normal; line-height: 48px; margin-bottom: 5%; 
      }
      
      @media screen and (max-width: 750px) {
        .main-container {
            margin: 5% 5%;
        }
      }
      `}</style>
      </React.Fragment>
    );
  } catch (error) {
    console.log("Dictionat Error ", error);
    return null;
  }
};
export default Dictionary;
