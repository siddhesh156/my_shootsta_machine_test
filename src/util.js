import axios from 'axios';

export const fetchData = async query => {
  const url = `https://api.dictionaryapi.dev/api/v2/entries/en/${query}`;
   let response = await axios.get(url)
   //console.log("response=> ",response);
  return response;
};
