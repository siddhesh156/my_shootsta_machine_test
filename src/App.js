import Dictionary from "./container/Dictionary";
import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';
function App() {
  return (
    <Dictionary initialState={"dog"} />
  );
}

export default App;
