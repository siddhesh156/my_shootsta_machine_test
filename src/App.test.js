import { render, screen, waitFor, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";
import "@testing-library/jest-dom";
const axios = require("axios");
import React, { useState } from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Dictionary from "../src/container/Dictionary";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { fetchData } from './util.js';

jest.mock("axios");

Enzyme.configure({ adapter: new Adapter() });

test("renders Dictionary", () => {
  render(<App />);
  const linkElement = screen.getByText(/Dictionary/i);
  expect(linkElement).toBeInTheDocument();
});

describe("Search Dictionary", () => {
  test("Search Field Elements should be present", () => {
    render(<App />);
    const submitButton = screen.getByText(/Submit/i);

    expect(submitButton).toBeInTheDocument();
    expect(submitButton).toBeEnabled();
  });
});

describe("Search Dictionary Success", () => {
  test("User can search ", async () => {
    render(<App />);
    const mockedResponse = {
              definitions: {
                definition:
                  "A mammal, Canis familiaris or Canis lupus familiaris, that has been domesticated for thousands of years, of highly variable appearance due to human breeding.",
                example: "The dog barked all night long.",
              }
    };

    axios.get.mockResolvedValue(mockedResponse);
    axios.get = jest.fn(() => mockedResponse);
    axios.get('https://api.dictionaryapi.dev/api/v2/entries/en/dog');
    expect(axios.get).toHaveBeenCalledWith('https://api.dictionaryapi.dev/api/v2/entries/en/dog'); 
    expect(axios.get).toHaveReturnedWith(mockedResponse);
    await expect(fetchData('dog')).resolves.toEqual(mockedResponse);
  });
});

describe("Search Dictionary Error", () => {
  test("User can search ", async () => {
    render(<App />);
    const mockedResponse = {
      message: "Sorry pal, we couldn't find definitions for the word you were looking for. You can try another word."
    };

    axios.get.mockResolvedValue(mockedResponse);
    axios.get = jest.fn(() => mockedResponse);
    axios.get('https://api.dictionaryapi.dev/api/v2/entries/en/agf');
    expect(axios.get).toHaveBeenCalledWith('https://api.dictionaryapi.dev/api/v2/entries/en/agf'); 
    expect(axios.get).toHaveReturnedWith(mockedResponse);
    await expect(fetchData('agf')).resolves.toEqual(mockedResponse);
  });
});

// describe("Search Dictionary Error", () => {
//   test("User can search ", async () => {
//     render(<App />);
//     const searchInput = screen.getByLabelText(/Search/i);
//     const submitButton = screen.getByText(/Submit/i);
//     expect(searchInput).toHaveValue("");
//     userEvent.type(searchInput, "agf");
//     expect(searchInput).toHaveValue("agf");
//     userEvent.click(submitButton);

//     await waitFor(() => screen.getByText(/Sorry pal/i));
//   });
// });

// test('setOpenIndex sets the open index state properly', () => {
//   const wrapper = mount(<Dictionary />)
//   expect(wrapper.state('word')).toBe("")
//   wrapper.instance().setWord("dog")
//   expect(wrapper.state('word')).toBe("dog")

// const query = "nirvana";
// const mockChange = jest.fn();
// searchInput.onChange = mockChange;
// fireEvent.change(searchInput, { target: { value: query } });
// })

// describe("mock api calls", () => {
  // const mockedResponse = {
    //   data: [
    //     {
    //       meanings: [
    //         {
    //           definitions: {
    //             definition:
    //               "A mammal, Canis familiaris or Canis lupus familiaris, that has been domesticated for thousands of years, of highly variable appearance due to human breeding.",
    //             example: "The dog barked all night long.",
    //           },
    //         },
    //       ],
    //     },
    //   ],
    //   status: 200,
    // };

//    test("mocking external endpoint in axios", () => {

//        // arrange
//        const mockedResponse = {data: "dog"}
//        axios.get.mockResolvedValue(mockedResponse)
//        //const app = require('../app.js')

//        // act
//       // app.getUserData()

//        // asserts
//        expect(axios.get).toHaveBeenCalled()
//        expect(axios.get).toHaveBeenCalledWith('https://api.dictionaryapi.dev/api/v2/entries/en/dog')

//await waitFor(() => screen.getByText(/Definitions/i))
//    })
// })
