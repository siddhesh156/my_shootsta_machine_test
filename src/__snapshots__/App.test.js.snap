// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Dictionary Success User can search  1`] = `ShallowWrapper {}`;
